src/components/App.jsx

```diff
@@ -1,25 +1,43 @@
 import React from 'react'
-// import { BrowserRouter } from 'react-router-dom'
+import { BrowserRouter as Router, Route } from 'react-router-dom'
 
-const App = () => (
-  <div>
-    <nav className="navbar is-dark has-shadow">Gist nav bar menu</nav>
-    <section> list of available gists </section>
-    <main className="container">
-      <section>Gist body
-        <div>Gist description</div>
-        <div>gist main
-          <nav>toolbar
-            <div>file name extension</div>
-            <div>writing options</div>
-          </nav>
-          <text>body of the gist</text>
-        </div>
-        <section>buttons</section>
-      </section>
-    </main>
-    <section>Footer</section>
-  </div>
-)
+import GistsList from './GistsList'
+import GistView from './GistView'
+import AboutView from './AboutView'
+import Layout from './Layout'
+import HomePage from './HomePage'
+
+
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { hasError: false }
+  }
+
+  componentDidCatch(error, info) {
+    this.setState({ hasError: true })
+    console.log(error, info)
+  }
+  render() {
+    if (this.state.hasError) {
+      return <h2>Something went wrong</h2>
+    }
+    return (
+      <main>
+        <Router>
+          <div className="container">
+            <Layout>
+              <Route exact path="/" component={HomePage} />
+              <Route exact path="/gists" component={GistsList} />
+              <Route path="/gist/:id" component={GistView} />
```

Comment:
why not `/gists/:id`?

Reply:
I thought that for a single gist, it would be better to have 'gist' instead of 'gists' :)